core/trino-main/src/main/java/io/trino/memory/TotalReservationOnBlockedNodesLowMemoryTaskKiller.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.trino.memory;
+
+import com.google.common.collect.ImmutableSet;
+import io.trino.execution.TaskId;
+import io.trino.operator.RetryPolicy;
+import io.trino.spi.QueryId;
+import io.trino.spi.memory.MemoryPoolInfo;
+
+import java.util.AbstractMap.SimpleEntry;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import static com.google.common.collect.ImmutableSet.toImmutableSet;
+
+public class TotalReservationOnBlockedNodesLowMemoryTaskKiller
+        implements LowMemoryKiller
+{
+    @Override
+    public Optional<KillTarget> chooseQueryToKill(List<QueryMemoryInfo> runningQueries, List<MemoryInfo> nodes)
```

Comment:
Should we rename to `chooseTargetToKill`?

Reply:
Sure - it was not accureate already before this PR but it makes sense to change it.